source/core/Events.cpp

```diff
@@ -45,6 +45,9 @@ UserEventRefNumManager UserEventRefNumManager::_s_singleton;
 typedef Int32 EventQueueID;
 enum { kNotAQueueID };
 
+// Monotonically incremented event number, so we can always tell which event is generated first even if timestamps are the same
+UInt32 EventData::_s_eventSequenceNumber = 0;
```

Comment:
Are we concerned about this number wrapping around when the value gets to Max number in the UInt32 range?

Reply:
This is actually handled -- the comparison of this value with previous values if done using a helper function EventTimeSeqCompare which handles wrapping.